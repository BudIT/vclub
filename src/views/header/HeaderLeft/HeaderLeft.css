.ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  display: flex;
  justify-content: flex-start;
  background-color: #FFD00B;
  flex: 1;
}

.a {
  color: #33363A;
  text-decoration: none;
  text-align: center;
  padding: 14px 16px;
  font-size: 22px;
  transition: 0.2s;
  display: inline-block;
  font-family: 'Rubik', helvetica, sans-serif;
}

.aCurrent {
  composes: a;
  text-decoration: underline;
}

.a:hover {
  background-color: #FFAA00;
}
