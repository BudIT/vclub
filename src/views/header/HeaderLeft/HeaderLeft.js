import React, { PropTypes, Children } from 'react';

import HeaderTab from './HeaderTab/HeaderTab';

import style from './HeaderLeft.css';

// changeRoom
function HeaderLeft(props) {
  // console.log("headerleft")
  // console.log(props);

  // return different style for tab if tab and room are the same
  // see tab with underline in ui

  const tabs = Children.map(props.children, (roomName, index) => (
    <HeaderTab
      key={index.toString()}
      {...props}
    >
      {roomName}
    </HeaderTab>
  ));

  return (
    <ul className={style.ul}>
      {tabs}
    </ul>
  );
}

HeaderLeft.propTypes = {
  // rooms: PropTypes.arrayOf(PropTypes.string).isRequired,
  // currentRoom: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,

  // getChangeRoomHandlers: PropTypes.arrayOf(PropTypes.func.isRequired).isRequired,

  // dispatch: PropTypes.func.isRequired,
};

export default HeaderLeft;
