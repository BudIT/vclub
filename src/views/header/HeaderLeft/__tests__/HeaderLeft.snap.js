/* eslint import/no-extraneous-dependencies: ["error", {"devDependencies": true}] */
/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import HeaderLeft from '../HeaderLeft';

const roomsNames = ['SHARING', 'VIDEO', 'CHAT', 'WHITEBOARD'];
const currentRoomName = 'VIDEO';

const user = {
  id: '1',
  name: 'User',
  master: true,
};

test('<HeaderLeft /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const rendered = renderer.create(
    <HeaderLeft
      currentRoomName={currentRoomName}
      roomsNames={roomsNames}
      user={user}
      dispatch={dispatchSpy}
    />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
