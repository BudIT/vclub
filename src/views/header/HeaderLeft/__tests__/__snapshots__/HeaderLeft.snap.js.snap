exports[`test <HeaderLeft /> renders correctly 1`] = `
<ul
  className="ul">
  <li>
    <button
      className="tab"
      onClick={[Function]}>
      SHARING
    </button>
  </li>
  <li>
    <button
      className="tabCurrent"
      onClick={[Function]}>
      VIDEO
    </button>
  </li>
  <li>
    <button
      className="tab"
      onClick={[Function]}>
      CHAT
    </button>
  </li>
  <li>
    <button
      className="tab"
      onClick={[Function]}>
      WHITEBOARD
    </button>
  </li>
</ul>
`;
